ecommerce/views/v0/__init__.py

```diff
@@ -84,3 +94,41 @@ def get_queryset(self):
             .select_related("content_type")
             .prefetch_related("purchasable_object")
         )
+
+
+class BasketViewSet(
+    NestedViewSetMixin, mixins.ListModelMixin, mixins.RetrieveModelMixin, GenericViewSet
+):
+    """API view set for Basket"""
+
+    serializer_class = BasketSerializer
+    permission_classes = [IsAuthenticated]
+    lookup_field = "user__username"
+    lookup_url_kwarg = "username"
+
+    def get_object(self, username=None):
+        basket, _ = Basket.objects.get_or_create(user=self.request.user)
+        return basket
+
+    def get_queryset(self):
+        return Basket.objects.filter(user=self.request.user).all()
+
+
+class BasketItemViewSet(
+    NestedViewSetMixin, ListCreateAPIView, mixins.DestroyModelMixin, GenericViewSet
+):
+    """API view set for BasketItem"""
+
+    serializer_class = BasketItemSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return BasketItem.objects.filter(basket__user=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+        basket = Basket.objects.get(user=request.user)
+
+        product_id = request.data.get("product")
+        product = Product.objects.get(id=int(product_id))
+        BasketItem.objects.create(basket=basket, product=product)
```

Comment:
This might be better as a `get_or_create` - returning a `201` if it's been created and a 200 otherwise - it also ideally would return the serialized basket item so that might mean moving this creation logic into the serializer's `create()` method instead to be cleaner since then the viewset would take care of the serialization as long as `create()` returns an instance.